cordapp/src/test/kotlin/com/negotiation/contracts/ModificationContractTests.kt

```diff
@@ -1,6 +1,81 @@
 package com.negotiation.contracts
 
+import com.negotiation.ProposalAndTradeContract
+import com.negotiation.ProposalState
+import net.corda.core.identity.CordaX500Name
+import net.corda.testing.core.TestIdentity
+import net.corda.testing.node.MockServices
+import net.corda.testing.node.ledger
+import org.junit.Test
+
 class ModificationContractTests {
+    private val ledgerServices = MockServices(listOf("com.negotiation", "net.corda.testing.contracts"))
+    private val alice = TestIdentity(CordaX500Name("alice", "New York", "US"))
+    private val bob = TestIdentity(CordaX500Name("bob", "Tokyo", "JP"))
+    private val charlie = TestIdentity(CordaX500Name("charlie", "London", "GB"))
+
+
+
+    @Test
+    fun `proposal transactions have exactly diffrent ammounts`() {
+        ledgerServices.ledger{
+            transaction {
+
+                command(listOf(alice.publicKey, bob.publicKey), ProposalAndTradeContract.Commands.Modify())
+                tweak {
+                    input(ProposalAndTradeContract.ID,ProposalState(1, alice.party, bob.party, alice.party, bob.party))
+                    output(ProposalAndTradeContract.ID, ProposalState(1, alice.party, bob.party, alice.party, bob.party))
+                    fails()
+                }
+                input(ProposalAndTradeContract.ID,ProposalState(1, alice.party, bob.party, alice.party, bob.party))
+                output(ProposalAndTradeContract.ID, ProposalState(2, alice.party, bob.party, alice.party, bob.party))
+                verifies()
+            }
+        }
+    }
+
+    @Test
+    fun `proposal transactions have only one output input and output`() {
```

Comment:
Should also check the input and output are proposal states.

Reply:
This hasn't been addressed.

Reply:
https://github.com/BCSTech-CordaTeam/negotiation-cordapp/blob/contractvalidations/cordapp/src/test/kotlin/com/negotiation/contracts/ModificationContractTests.kt

Can you please check here...
@Test
    fun `proposal transaction's input and output should be of type ProposalState`() {
        ledgerServices.ledger{
            transaction {
                input(ProposalAndTradeContract.ID,ProposalState(1, alice.party, bob.party, alice.party, bob.party))
                output(ProposalAndTradeContract.ID, ProposalState(2, alice.party, bob.party, alice.party, bob.party))
                command(listOf(alice.publicKey, bob.publicKey), ProposalAndTradeContract.Commands.Modify())
                tweak {
                    input(ProposalAndTradeContract.ID,DummyState())
                    fails()
                }
                tweak {
                    output(ProposalAndTradeContract.ID, DummyState())
                    fails()
                }
               verifies()
            }
        }
    }